sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -43,10 +70,387 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
+    const application = await this.applicationRepo.findOne({
+      select: {
+        id: true,
+        student: {
+          id: true,
+          studentRestrictions: {
+            id: true,
+            restriction: {
+              id: true,
+              restrictionCode: true,
+              actionType: true,
+            },
+            applicationRestrictionBypasses: {
+              id: true,
+              studentRestriction: { id: true },
+              isActive: true,
+            },
+            createdAt: true,
+          },
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+          },
+        },
+      },
+      relations: {
+        student: {
+          studentRestrictions: {
+            restriction: true,
+            applicationRestrictionBypasses: {
+              studentRestriction: true,
+            },
+          },
+        },
+        currentAssessment: { offering: true },
+      },
+      where: {
+        id: applicationId,
+        student: {
+          studentRestrictions: {
+            isActive: true,
+            restriction: {
+              actionType: ArrayOverlap([
+                RestrictionActionType.StopFullTimeBCFunding,
+                RestrictionActionType.StopFullTimeDisbursement,
+                RestrictionActionType.StopPartTimeDisbursement,
+              ]),
+            },
+          },
+        },
+      },
+    });
+
+    const allowedRestrictionActions =
+      application.currentAssessment.offering.offeringIntensity ===
+      OfferingIntensity.fullTime
+        ? [
+            RestrictionActionType.StopFullTimeBCFunding,
+            RestrictionActionType.StopFullTimeDisbursement,
+          ]
+        : [RestrictionActionType.StopPartTimeDisbursement];
+
+    const bypassedStudentRestrictionIds =
+      application.student.studentRestrictions
+        .flatMap(
+          (studentRestriction) =>
+            studentRestriction.applicationRestrictionBypasses,
+        )
+        .filter((bypass) => bypass.isActive)
+        .map((bypass) => bypass.studentRestriction.id);
+
+    const studentRestrictionsThatAreNotBypassed =
+      application.student.studentRestrictions.filter(
+        (studentRestriction) =>
+          !bypassedStudentRestrictionIds.includes(studentRestriction.id),
+      );
+
+    return studentRestrictionsThatAreNotBypassed
+      .filter((studentRestriction) =>
+        allowedRestrictionActions.some((actionType) =>
+          studentRestriction.restriction.actionType.includes(actionType),
+        ),
+      )
+      .map((studentRestriction) => ({
+        studentRestrictionId: studentRestriction.id,
+        restrictionCode: studentRestriction.restriction.restrictionCode,
+        studentRestrictionCreatedAt: studentRestriction.createdAt,
+      }))
+      .sort((a, b) => a.restrictionCode.localeCompare(b.restrictionCode));
```

Comment:
👍 This sort looks good to me, although there is no specific AC needed to sort the results.